bodhi/tests/server/services/test_updates.py

```diff
@@ -5028,3 +5028,169 @@ def test_unexpected_exception(self, log_exception, waive_test_results, *args):
         self.assertEquals(res.json_body['errors'][0]['description'],
                           u'IOError. oops!')
         log_exception.assert_called_once_with("Unhandled exception in waive_test_results")
+
+    @mock.patch.dict(config, [('test_gating.required', True)])
+    @mock.patch('bodhi.server.models.waiverdb_api_post')
+    @mock.patch('bodhi.server.models.greenwave_api_post')
+    def test_waive_test_results_1_unsatisfied_requirement(
+            self, greenwave_api_post, waiverdb_api_post, *args):
+        """Ensure that we get an error if trying to waive test results of a locked update"""
```

Comment:
The comment says the update is locked, but the name and body seem to be about there being one unsatisfied requirement.

Reply:
I'll adjust all the docstrings